### DISCUSSION TITLE
Magick drops bit depth from 24bit to 8 bit on an all-black image - how to force it to maintain 24 bit?

### DISCUSSION THREAD

**Discussion by creinemann** [ImageMagick/ImageMagick#2710] [Help] [answered]
So I am processing hundreds of satellite images with this line in my script:

`for /r "F:\Satellite Imagery\goes16\fd\CUSTOMLUT\" %%a in (*.jpg) do "C:\Program Files\ImageMagick-7.0.9-Q16\magick.exe"  mogrify -path ../COMPOSITEFILE/compositeFC -filter Triangle -define filter:support=2 -thumbnail 2000 -unsharp 0.25x0.25+8+0.065 -dither None -posterize 136 -quality 82 -define jpeg:fancy-upsampling=off -define png:compression-filter=5 -define png:compression-level=9 -define png:compression-strategy=1 -define png:exclude-chunk=all -interlace none -colorspace sRGB -strip  -gravity south -stroke none -pointsize 24 -fill white -font Ailerons -annotate +0+0 GOES_16_FC_and_Channel_15 "%%~a"`

This resizes, optimizes, adds text, and moves the original file into a new folder, but on night time passes when the image is, or nearly is, all black, Magick will drop the bit depth on just those few images from 24 bit to 8 bit, causing issues with color later on in my script when I layer different data sets together. EDIT- All images start at 96 DPI and 24-bit

Is there a way to force magick to only create 24-bit images?

Example of output, circled image is an example of the one  that gets dropped to 8 bit
[https://imgur.com/a/V2UPNnG](https://imgur.com/a/V2UPNnG)

**Comment by fmw42** [comment]
It is probably not the bit depth but that you need more gain. But if you want 24-bit color then do PNG24:output.png

**Comment by snibgo** [comment, marked as answer]
> ... when the image is, or nearly is, all black, Magick will drop the bit depth on just those few images from 24 bit to 8 bit, ...

I'm not sure what that means. IM does not have a bit depth of 24 bits. The number of bits per channel per pixel is 8 or 16 or 32 or 64. Images have one or more channels.

Your inputs and outputs are JPG. Your PNG options will do nothing.

Perhaps your very dark images are monochrome, so they need only one channel, to record 8-bit gray. To make JPG outputs have three channels, add "-type TrueColor".

**Reply by creinemann** [reply]
This did the trick, Thank you, and yes the -define:png served no purpose- I copied the line of script form another part of my overall batch file that does make use of .png's
